packages/instrumentation-socket.io/src/socket.io.ts

```diff
@@ -22,6 +22,12 @@ export class SocketIoInstrumentation extends InstrumentationBase<Io> {
     protected override _config!: SocketIoInstrumentationConfig;
 
     constructor(config: SocketIoInstrumentationConfig = {}) {
+        if (!Array.isArray(config.emitIgnoreEventList)) {
+            config.emitIgnoreEventList = [];
+        }
+        if (!Array.isArray(config.onIgnoreEventList)) {
+            config.onIgnoreEventList = [];
+        }
```

Comment:
Like on checking the config input. If we do it here, I would also add it to the `setConfig` function, and then we can omit checking if it's an array when we use it in the patched functions.